doc/conf.py

```diff
@@ -49,6 +57,32 @@
     "sphinx_reredirects",
 ]
 
+MSG_TYPES_DOC = {k: v if v != "info" else "information" for k, v in MSG_TYPES.items()}
```

Comment:
Can't we use https://documatt.gitlab.io/sphinx-reredirects/usage.html#source-wildcards here? I only just found this in the documentation.

Reply:
It does not work, we need a file to exists.
```python
def get_wildcard_messages_redirect() -> dict[str, str]:
    result = {}
    for category in MSG_TYPES_DOC.values():
        before = f"messages/{category}/*"
        result[before] = f"user_guide/messages/{category}/$source.html"
    return result
```
Result:
```
WARNING: No documents match to 'messages/information/*' redirect.
WARNING: No documents match to 'messages/convention/*' redirect.
WARNING: No documents match to 'messages/refactor/*' redirect.
WARNING: No documents match to 'messages/warning/*' redirect.
WARNING: No documents match to 'messages/error/*' redirect.
WARNING: No documents match to 'messages/fatal/*' redirect.
```

Reply:
I'm trying to make the redirect by specifying everything works. If this does not work I'm thinking about:
- Having the file on the repository just to be able to redirect ?
- Not redirecting at all, after all the message have not been available for very long how many link would really be broken ?

Reply:
I'm trying to figure out RTD redirects. It "should" work. I'll give an update if I give up 😄

Reply:
Using ``'messages/error/invalid-length-returned': 'user_guide/messages/error/invalid-length-returned.html'``, ``messages/error/invalid-length-returned.html`` is redirected to ``messages/error/user_guide/messages/error/invalid-length-returned.html`` it's unexpected. Any idea for a hack ? Maybe I should fork the lib and "fix" it ?

Reply:
Okay, I got the RTD redirects working.

See:
https://pylint--6628.org.readthedocs.build/en/6628/messages/fatal/astroid-error.html

Reply:
Hmm good job, as the code did not change I suppose you used RTD interface? We might as well do everything in interface and remove the dependency to avoid confusion ?

Reply:
Yeah, you can check the `Redirects` tab to see what I did.

I'm opening an issue at RTD as we speak because the process is quite difficult and doesn't allow us all the freedom we want. I would suggest doing all manual redirects with the extension and doing the global redirects for directories through RTD (there are some limitations to it).

Would you mind cleaning up the PR and removing the unnecessary code from `conf.py`?

Reply:
Sure thing ! I did not want to use the interface at all (because only maintainer have access) but whatever work 😄 Also only maintainer should be confident enough to redirect entire directories in the doc imo. And I already took an unreasonable amount of time to make this 0.0.1 lib work

Reply:
See https://github.com/readthedocs/readthedocs.org/issues/9240.

Yeah, I don't really understand why this is not easier. It seems like something more projects would have run into already.
Anyway, for now I think this solution works. Although we'll never be sure until this gets merged and we're live.

Reply:
> Although we'll never be sure until this gets merged and we're live.

I'm pretty confident it will work 😄